### DISCUSSION TITLE
LSP Should report Not found as a warning and allow you to create the note as a code action

### DISCUSSION THREAD

**Discussion by MarcCoquand** [zk-org/zk#370] [Ideas]
### Verify

- [X] I searched the existing discussions for similar ideas

### Share your idea or feature request

Heya!

Oftentimes when I write notes, I will often link to notes that I know don't exist so I can go back and create them later. In Obsidian, linking to a note that doesn't exist will lead to a gray link which when pressed would create the note. 

I think it'd be cool if the LSP didn't report a note missing as an error but rather a warning, and that there was an LSP code action for creating a new note with the name.

**Comment by mcDevnagh** [comment]
There already is a configuration option for dead links, so you can set it to warning yourself:
https://github.com/zk-org/zk/blob/main/docs/config-lsp.md#diagnostics

An LSP Action to create a new note would be a good improvement. I'll put in a Feature Request. We aren't adding new features as of right now (see [README](https://github.com/zk-org/zk/blob/main/README.md#news-we-are-in-maintenance-mode)), but I would definitely want this in the next minor release.

**Comment by MarcCoquand** [comment]
> There already is a configuration option for dead links, so you can set it to warning yourself:
> https://github.com/zk-org/zk/blob/main/docs/config-lsp.md#diagnostics

Ah I didn't see this, thank you for sharing the link.

> An LSP Action to create a new note would be a good improvement. I'll put in a Feature Request. We aren't adding new features as of right now (see [README](https://github.com/zk-org/zk/blob/main/README.md#news-we-are-in-maintenance-mode)), but I would definitely want this in the next minor release.

Great!
